### PR TITLE
Release branches and slack notifications

### PR DESCRIPTION
**Pull request by mattcfox**
Changes:
 - Added the `create_release_branch` option that will create a release branch in addition to a tag. This is desired because the default GH actions UI can only target branches
 - New slack action to announce deployments

Closes:
 - https://hotelengine.atlassian.net/browse/SUP-1152
 - https://hotelengine.atlassian.net/browse/SUP-1151
 - https://hotelengine.atlassian.net/browse/SUP-1147

### CODE REVIEW

File: sls_deploy/slack_message/action.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                      "text": "View this action"
+                    },
+                    "url": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+                  }
+                ]
+              }
+            ]
+          }
+        SLACK_WEBHOOK_URL: ${{ inputs.slack_webhook }}
+    # - uses: slackapi/slack-github-action@v1.15.0
```
Comment (claytonweller) [inline sls_deploy/slack_message/action.yml:72]: Is there reason we're keeping this?

**Review by claytonweller — APPROVED** [review]
I'm not super familiar with what exactly is going on here. So I'm a ✅. Just one little question about the commented code below.

File: sls_deploy/slack_message/action.yml (hunk 1 above)
Comment (mattcfox) [reply sls_deploy/slack_message/action.yml:72]: Nope, that shouldn't be in there. Removing.

**Review by he-dslieker — APPROVED** [review]

**Review by zjromani — APPROVED** [review]
This looks great. I'll wait for the ruby folks to review before merging just to be safe (since we can already deploy to prod without this change)

Also, if anyone is interested, [this](https://hotelengine.slack.com/archives/C01DJT4C0AZ/p1630624007003800) is what the message will look like on deploy with this change.

**Review by claytonweller — APPROVED** [review]

File: tag/action.yml [hunk 2]
```diff
@@ -30,6 +34,7 @@ runs:
       env:
         SLACK_API_TOKEN: ${{ inputs.slack_api_token }}
         SLACK_CHANNEL_ID: ${{ inputs.slack_channel_id }}
+        CREATE_RELEASE_BRANCH: ${{ inputs.create_release_branch }}
```
Comment (dalexj) [inline tag/action.yml:37]: lets just go ahead and always do this. dont think we need to have a case where we dont

Comment (zjromani) [reply tag/action.yml]: ✅ updated

Comment (zjromani) [reply tag/action.yml]: I think we want to guard in here now for only releasing the `main` branch like we were doing in `supply` [here](https://github.com/HotelEngine/supply/pull/93/files#diff-b423e5dfbb1b554bfdca2fef78d85aa9b5bd507c175fbf712730fa3315fbf729R22)

**Review by dalexj — APPROVED** [review]

**Pull request merged**